### PR TITLE
Scripts/Northrend: fix compile errors in Trial of the Champion

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

-  Fix Compile erros show in Travis: https://travis-ci.org/TrinityCore/TrinityCore/builds/297437923?utm_source=github_status&utm_medium=notification

**Target branch(es):**
- [x] 3.3.5-trial_of_the_champion 
- [ ] 3.3.5
- [ ] master

### CODE REVIEW

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp [hunk 1]
```diff
@@ -1276,11 +1276,11 @@ class npc_announcer_toc5 : public CreatureScript
         return seen;
     }
 
-    bool OnGossipHello(Player* player, Creature* creature) override
+    bool GossipHello(Player* player) override
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: you have to move this inside the npc's ai

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp [hunk 2]
```diff
@@ -1243,6 +1243,102 @@ class npc_announcer_toc5 : public CreatureScript
...
+
+        bool GossipSelect(Player* player, uint32 /*menuId*/, uint32 gossipListId) override
+        {
+            uint32 const action = player->PlayerTalkClass->GetGossipOptionAction(gossipListId);
+            ClearGossipMenuFor(player);
+            CloseGossipMenuFor(player);
+            switch (action)
+            {
+                case GOSSIP_ACTION_INFO_DEF + 1:
+                    ENSURE_AI(npc_announcer_toc5::npc_announcer_toc5AI, me->AI())->StartEncounter();
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: these casts seems useless

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp [hunk 3]
```diff
@@ -1243,131 +1243,132 @@ class npc_announcer_toc5 : public CreatureScript
...
+            CloseGossipMenuFor(player);
+            switch (action)
             {
-                // if everyone from your group have completed one of the Trial of the Champion achievements, you have option to skip the event
-                // maybe not the correct way to do it but I couldn't figure out better
-                if (!plr->HasAchieved(4298) /* Heroic ToC (alliance) */ && !plr->HasAchieved(3778) /* Normal ToC (horde) */ && !plr->HasAchieved(4297) /* Heroic ToC (horde) */ && !plr->HasAchieved(4296) /* Normal ToC (alliance) */)
-                {
-                    seen = false;
+                case GOSSIP_ACTION_INFO_DEF + 1:
+                    ENSURE_AI(npc_announcer_toc5::npc_announcer_toc5AI, me->AI())->StartEncounter();
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: these casts seem useless

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp [hunk 4]
```diff
@@ -25,6 +25,8 @@ EndScriptData */
 #include "Player.h"
 #include "ScriptMgr.h"
 #include "ScriptedCreature.h"
+#include "ScriptMgr.h"
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp]: duplicate

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp [hunk 5]
```diff
@@ -26,6 +26,7 @@ EndScriptData */
 #include "ScriptMgr.h"
 #include "ScriptedCreature.h"
 #include "ScriptedEscortAI.h"
+#include "ScriptMgr.h"
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp]: also duplicate

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp (hunk 3 above)
Comment (Jildor) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: Well, I'm fixing compile errors, all these casts for you 😘

Comment (joschiwald) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: that is part of fixing compile errors

Comment (Jildor) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: Now  travis  seems happy....
And I don't know very  well how change these casts xD

Comment (joschiwald) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: just remove `ENSURE_AI(npc_announcer_toc5::npc_announcer_toc5AI, me->AI())->`

**Comment by tkrokli** [comment]
Good job fixing the compile errors in Travis. 👍
Anything else you think is needed before merge?

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp (hunk 3 above)
Comment (tkrokli) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: OK, so it should work with only `StartEncounter();` on this line? 😅

Comment (joschiwald) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: yes

Comment (Jildor) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp]: Now it's ok @joschiwald ??

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp [hunk 6]
```diff
@@ -402,7 +402,7 @@ class boss_black_knight : public CreatureScript
                 {
                     Creature* ghoul = ObjectAccessor::GetCreature(*me, *itr);
                     if (ghoul && ghoul->IsAlive() && !ghoul->HasAura(SPELL_GHOUL_EXPLODE))
-                        ENSURE_AI(npc_risen_ghoul::npc_risen_ghoulAI, ghoul->AI())->DoExplode();
```
Comment (joschiwald) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp]: not here

Comment (Jildor) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp]: oks, done

**Comment by Jildor** [comment]
If @joschiwald approve last changes I think I finished xD

**Review by ghost — APPROVED** [review]
Looks Good To Me™

**Comment by tkrokli** [comment]
Going to merge this as soon as [AppVeyor has finished](https://ci.appveyor.com/project/DDuarte/trinitycore/build/1.0.7453) compiling.
Mostly because we need to have the compile errors solved.

**Comment by Jildor** [comment]
Seems AppVeyor slower than mclaren honda :grinning:

Maybe tomorrow

**Comment by tkrokli** [comment]
Ah, what the heck. No real need to wait for AppVeyor to finish, I just thought it would be nice to see it finish to have an "all green" status on this PR before merge. I will merge it soon anyway. :-P

**Comment by ConradBunton** [comment]
Appled and the other make good job but syntax was changed and merge gone wrong... Until today.
Thx Jildor, no error left at the compil. I'll tell you if I see a problem.

**Comment by ConradBunton** [comment]
Ok, I tried it.
1st problem : In the intro, the door open but npc don't take their WP. Same problem with the 2 others groups.

**Comment by Jildor** [comment]
@ConradBunton can you test after @ccrs latest waypoint movement changes?
https://github.com/TrinityCore/TrinityCore/commit/e10d7dd45cfc073c791cd5e5f3f964d14b37df3d

**Comment by ConradBunton** [comment]
I'll try and tell you.

**Comment by Furrazzlhf** [comment]
Yes and if i can up the conversation , same problem like Conrad and

* The 3 bosses doesn't use spells , only auto attack.

**Comment by Jildor** [comment]
Meybe @tkrokli can free @Keader and @sirikfoll from the basement and they can make a little cleanup 🙏

**Comment by tkrokli** [comment]
`*evil laugh*` /JK

To be perfectly honest, I don't have much influence over neither Keader, sirikfoll nor any other TC member.
Real life is a much bigger hinder for all of us to even think about how we influence each other.
<small><sub>("How To Win Friends And Influence Enemies")</sub></small> ;-)

**Comment by ConradBunton** [comment]
Finally, TOC5 is just a dead branch. It just will stay like that...

**Comment by Aokromes** [comment]
Like i said on the previous PR the best way to get it merged is split it on multiple PRs it makes more easy to review and focus on needed things.

**Comment by tkrokli** [comment]
Yes, please make some new PRs to the [`TrinityCore:3.3.5-trial_of_the_champion`](https://github.com/TrinityCore/TrinityCore/tree/3.3.5-trial_of_the_champion) branch, preferably small ones which will bring it closer to the [main `3.3.5`](https://github.com/TrinityCore/TrinityCore/tree/3.3.5) branch so the improved content can be merged into `3.3.5` later.

**Pull request merged**